src/nimib.nim

```diff
@@ -189,14 +189,27 @@ when moduleAvailable(karax/kbase):
     nbRawHtml: "<div id=\"" & rootId & "\"></div>"
     nbKaraxCodeBackend(rootId, args)
 
-template nbJsShowSource*(message: string = "") =
+template nbJsShowSource*(message: string = "") {.deprecated: "Use nbCodeDisplay instead".} =
   nb.blk.context["js_show_nim_source"] = true
   if message.len > 0:
     nb.blk.context["js_show_nim_source_message"] = message
 
-template nbCodeToJsShowSource*(message: string = "") {.deprecated: "Use nbJsShowSource instead".} =
+template nbCodeToJsShowSource*(message: string = "") {.deprecated: "Use nbCodeDisplay instead".} =
   nbJsShowSource(message)
 
+template nbCodeDisplay*(tmplCall: untyped, body: untyped) =
+  ## display codes used in a template (e.g. nbJsFromCode) after the template call
+  tmplCall:
+    body
+  newNbCodeBlock("nbCode", body):
+    discard
+
+template nbCodeAnd*(tmplCall: untyped, body: untyped) =
```

Comment:
What if the user wants to run the code in another block than `nbCode`, eg. `nbCodeInBlock`? I think it would make sense that `nbCodeAnd` takes in two `tmplCall` so the user can choose for themselves. Plus, it becomes more general. I don't know how this works out with nbCodeAsInSource though :sweat_smile: So maybe it isn't possible :shrug:

Reply:
for this I think the quicker solution would be to provide with a `nbCodeInBlockAnd`.

In principle we could add a second template and do `nbThisAndThat(nbCodeInBlock, nbJsFromSource)` but this would not compose with `codeAsInsource` (which is actually the full purpose here).